amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/parser/spec/jsonschema/AstFinder.scala

```diff
@@ -0,0 +1,151 @@
+package amf.plugins.document.webapi.parser.spec.jsonschema
+
+import amf.core.Root
+import amf.core.client.ParsingOptions
+import amf.core.model.document.{BaseUnit, ExternalFragment, Fragment, RecursiveUnit}
+import amf.core.parser.errorhandler.ParserErrorHandler
+import amf.core.parser.{EmptyFutureDeclarations, JsonParserFactory, ParsedReference, ParserContext, Reference, SchemaReference, SyamlParsedDocument}
+import amf.plugins.document.webapi.contexts.WebApiContext
+import amf.plugins.document.webapi.contexts.parser.oas.JsonSchemaWebApiContext
+import amf.plugins.document.webapi.contexts.parser.raml.Raml08WebApiContext
+import amf.plugins.document.webapi.parser.spec.common.YMapEntryLike
+import amf.plugins.document.webapi.parser.spec.toOasDeclarations
+import amf.validations.ParserSideValidations.UnableToParseJsonSchema
+import org.yaml.model.{YDocument, YMap, YNode}
+import org.yaml.parser.{YParser, YamlParser}
+
+object AstFinder {
```

Comment:
Seems this object has more responsibilities than finding nodes in an AST

Reply:
Added a commit that will solve this issue